source-kafka/src/configuration.rs

```diff
@@ -26,7 +28,19 @@ pub struct Configuration {
 }
 
 impl Configuration {
-    pub fn parse<R: Read>(reader: R) -> Result<Configuration, Error> {
+    pub fn brokers(&self) -> String {
+        self.bootstrap_servers
+            .iter()
+            .map(|url| url.to_string())
```

Comment:
Good catch. Interestingly, `itertools` isn't even a transitive dependency yet. I sort of expected it to be pulled in by something, but given that it isn't yet, I'll leave this be for now. 

FWIW, I don't think these allocations aren't really have any meaningful impact since they only happen at boot time. If we want to try and excise some allocations, we have better targets in Topic names that are being repeatedly cloned.